microsoft/testsuites/core/storage.py

```diff
@@ -0,0 +1,203 @@
+        entry with mount point `/' in the `mount` command, and the other is
+        resource disk.
+        3. Verify that "/dev/<resource_disk_partition> <mount_point>` entry is
+        present in `/etc/mtab` file.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def verify_resource_disk_mtab_entry(self, log: Logger, node: RemoteNode) -> None:
+        resource_disk_mount_point = self._get_data_disk_mount_point(log, node)
+        os_disk_partition = self._get_os_disk_partition(log, node)
+        if os_disk_partition == "sda":
+            mntresource = "/dev/sdb1 " + resource_disk_mount_point
+        else:
+            mntresource = "/dev/sda1 " + resource_disk_mount_point
+
+        output = node.tools[Cat].run("/etc/mtab").stdout
+        if mntresource in output:
+            log.debug("ResourceDisk entry is present.")
+        else:
+            raise LisaException("ResourceDisk entry is not present.")
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that the swap is correctly configured on the VM.
+        Steps:
+        1. Check if swap file/partition is configured by checking the output of
+        `swapon -s` and `lsblk`.
+        2. Check swap status in `waagent.conf`.
+        3. Verify that values in step 1 and step 2 are equal.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def verify_swap(self, log: Logger, node: RemoteNode) -> None:
+        swapon_output = node.tools[SwapOn].run("-s", sudo=True).stdout
+        lsblk_output = node.tools[Lsblk].run().stdout
+        _is_swap_enabled = self._is_swap_enabled(log, node)
+
+        if (
+            ("swap" in swapon_output) or ("SWAP" in lsblk_output)
+        ) and not _is_swap_enabled:
+            raise LisaException(
+                "Swap is disabled in waagent.conf but swap partition/files found."
+            )
+
+        if (
+            ("swap" not in swapon_output) and ("SWAP" not in lsblk_output)
+        ) and _is_swap_enabled:
+            raise LisaException(
+                "Swap is enabled in waagent.conf but no swap partition/files found."
+            )
+
+    def _get_wala_conf_path(self, node: RemoteNode) -> str:
+        if isinstance(node.os, CoreOs):
+            return "/usr/share/oem/waagent.conf"
+        elif node.os.information.vendor == "clear-linux-os":
+            return "/usr/share/defaults/waagent/waagent.conf"
+        else:
+            return "/etc/waagent.conf"
+
+    def _get_data_disk_mount_point(
+        self,
+        log: Logger,
+        node: RemoteNode,
+    ) -> str:
+        if node.shell.exists(
+            PurePosixPath("/var/log/cloud-init.log")
+        ) and node.shell.exists(PurePosixPath("/var/lib/cloud/instance")):
+            log.debug("Data handled by cloud-init.")
```

Comment:
What means "data" in the log? Should it be "provisioning"?

Reply:
This mean that cloud-init is handling disk mounting